### ISSUE TITLE
Visualization of followsFrom relation

### ISSUE THREAD

**Issue by chrvo** [jaegertracing/jaeger-ui#662]
## Requirement - what kind of business use case are you trying to solve?

We have multiple arbitration services that are called by multiple other service. All incoming messages are aggregated under certain conditions in a defined time window:

                                 +----+
                                 |    | Timer
                                 v    |
                           +-------+  |
                           |       |  |
                           |   A   +--+
                           |   G   |
     Service A +---------> |   G   |
                           |   R   |    Result
     Service B +---------> |   E   | ------------->
                           |   G   |
     Service C +---------> |   A   |
                           |   T   |
                           |   E   |
                           |       |
                           +-------+

We generate a span for the result with links to the spans from all aggregated requests (OpenTelemetry Links which are exported als Jaeger `followsFrom`).

## Problem - what in Jaeger blocks you from solving the requirement?

At moment, we have a bunch of different independent traces in the Jaeger which is not optimal for tracing the behavior.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

We generate a span for the result with links to the spans from all aggregated requests (OpenTelemetry Links which are exported als Jaeger `followsFrom`).
At moment, we have a bunch of different independent traces in the Jaeger, which is quiet suboptiomal.

As the followsFrom links are not visualized yet, we brainstormed what visualization would help us.
1. A link/followsFrom section in the span properties
The section should be on the same level as Tags, Process and Logs and should contain HTML links to the spans that a) follows from and b) are followed from this span
![jaeger_links](https://user-images.githubusercontent.com/58291334/99944903-f6cc3880-2d73-11eb-93c0-5c66ba154132.png)

2. All linked traced should appear in the same trace view. With different "root nodes" in the Traceview This could be optionally with a "show linked traces" checkbox to enable it.
![mt](https://user-images.githubusercontent.com/58291334/99944872-e916b300-2d73-11eb-9f09-359396be6167.png)

3. Highlight linked (both ways) span on mouse over/selection events.

**Comment by yurishkuro** [comment]
+1 for the Links section - I thought we already implemented that. cc @rubenvp8510 

However, there is another visual indication when you have a foreign SpanReference:

![image](https://user-images.githubusercontent.com/3523016/100003259-c6949200-2d93-11eb-9972-87be70cace7f.png)

**Comment by rubenvp8510** [comment]
I remember I've already implemented something similar in this PR: https://github.com/jaegertracing/jaeger-ui/pull/477

There should be a references accordion, but it will be showed only when the span has more than one reference, assuming that the first reference is the parent, and other ones could be child_off or follows_from. ( See on https://github.com/jaegertracing/jaeger-ui/pull/477/files#diff-847e772b786d5a6cdf1b72741721b570051516ac87338ad18bb6dc08472e9f56R143)

**Comment by cg110** [comment]
I'm not sure if this is also the same issue I'm having, if a span is has a Follows_From reference then it gets a link etc to the span it follows from, which is good.

However, that followed span doesn't seem have any indication that it's been followed. So if you look at that span there's nothing to link it to the spans that follows.  (I don't know if this is a backend limitation or UI issue)

It'd be great to have a link so you can find the "followers", or even better be able to have the multiple traces so the followed and followers can be on one.

In our case we've some async calls going on, and we get a callback once complete, in which we know the original span, and we either make that callback be a child (so all on one view) or follow_from the original span.  Follow_from feels the right thing, as the callback should be it's own trace, but it looks better in the ui as a child.

**Comment by Sushisource** [comment]
Any update on this? I ran into this today and traces that are linked together are pretty much useless because there's no way to see which spans follow from the span you're looking at, only backlinks are represented in the UI.

Having a unified view of all spans that followed (ideally transitively as well) from some given span is I would suspect a use case that many, many people would care about.

**Comment by ndrwrbgs** [comment]
@yurishkuro what conditions make this experience surface?



> +1 for the Links section - I thought we already implemented that. cc @rubenvp8510
> 
> However, there is another visual indication when you have a foreign SpanReference:
> 
> ![image](https://user-images.githubusercontent.com/3523016/100003259-c6949200-2d93-11eb-9972-87be70cace7f.png)

**Comment by yurishkuro** [comment]
When a span has span reference pointing to another trace.

**Comment by jkowall** [comment]
Should we close this out now @yurishkuro ?

**Comment by yurishkuro** [comment]
Yes, if OP has no further feedback

**Comment by ndrwrbgs** [comment]
I think there's enough context and +1 provided by others its still an open concern.

It turns out the link ui only shows when the span has more than one link, so in the case of follows from, you'd have to navigate to the child span manually and then it wouldn't have the pictured UI, but would have link inside the expanded view.

I believe OPs point was that there's no link back. I had to manually change how OpenTelemetry was emitting data, take independent control of the SDK and emit extra spans so that I could make a normal follows from, then a fake tie-between span that parented up to the follows from and the parent to get it to show in the UI.

Basically, right now, a follows_from span does not appear in the trace that it followed from (the causal span)

**Comment by arun0009** [comment]
Is there a way to connect `followsfrom` together for viewing? Basically if I search for `parent` Trace Id I want to get results from anything that has been linked with `followsfrom`?

e.g. :
Lambda sending something to SQS (first flow) - I see a trace here with few spans.
Lambda consuming from SQS (second flow) - I see a trace here with few spans but also a `link` followsfrom "first flow".

How can I get both of them to show me so I can look at the complete flow? (using opentelemetry auto instrumentation js).

<img width="1728" alt="1" src="https://github.com/jaegertracing/jaeger-ui/assets/3824444/160c5173-dfb7-4798-b193-4375329a8f79">


<img width="1727" alt="image" src="https://github.com/jaegertracing/jaeger-ui/assets/3824444/85ce6007-fd52-45ae-a841-49a4e7488b65">

**Comment by SwanHtetAung** [comment]
If it is a different traceID, you will have to make an attribute that tells how follow from children it has. After that, change jaeger-query UI so that, if it sees that value, it will also query the Jaeger backend and properly link the complete flow together.

Cause if I am correct, the Jaeger parent span will not know how many FollowFrom children it has. In short, I think there should be another attribute in spanContext that tell the current span followFrom children.
